crates/rolldown/tests/esbuild/loader/require_custom_extension_data_url/artifacts.snap

```diff
@@ -15,7 +15,7 @@ var test_ns, test_default;
 var init_test = __esmMin(() => {
 	test_ns = {};
 	__export(test_ns, {default: () => test_default});
-	test_default = "data:text/plain;charset=utf-8,a%5Cx00b%5Cx80c%5CxFFd";
+	test_default = "data:text/plain;charset=utf-8,a\\x00b\\x80c\\xFFd";
```

Comment:
```js
var __getOwnPropNames = Object.getOwnPropertyNames;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};

// a/foo.custom
var require_foo = __commonJS({
  "a/foo.custom"(exports, module) {
    module.exports = "data:text/plain;charset=utf-8,a\\x00b\\x80c\\xFFd%0A";
  }
});

// a/entry.js
var foo = require_foo();
console.log(foo);
//# sourceMappingURL=index.js.map
```

Reply:
The percent decoding may be required according to [here](https://github.com/rolldown/rolldown/pull/1406#issuecomment-2212384575), either using the `urlencoding` or using the `percent decode` function.

Reply:
percent decode is required according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs#datahello2c20world21).

Reply:
for this example, it did not specify `text/plain` and charset, https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs#datahello2c20world21